### ISSUE TITLE
weave-kube: Make a v1.6 compatible yaml file

### ISSUE THREAD

**Issue by luxas** [weaveworks/weave#2777]
In v1.6 two APIs are graduated to beta that affects Weave Net, namely: RBAC and Taints and Tolerations.
Also, RBAC will be enabled by default on most deployments, which doesn't allow any access to the api server when using the default kube-system ServiceAccount.
We have to provide RBAC rules for v1.6 clusters. We can provide the rules regardless of whether the user have RBAC enabled or not, but we can't provide it for v1.5 clusters given the alpha  => beta change.

We're also using the alpha toleration annotation which becomes a field on `PodSpec` in v1.6, and the taint that's used for the master is also subject to change.

x-ref: https://github.com/kubernetes/kubeadm/issues/143 and https://github.com/kubernetes/kubeadm/issues/150

cc @errordeveloper @lukemarsden @bboreham

**Comment by luxas** [comment]
I looked into this and found out what the RBAC rules should look like:
```
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: weave-net
rules:
- apiGroups:
  - ""
  resources:
  - pods
  - namespaces
  - nodes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - extensions
  resources:
  - networkpolicies
  verbs:
  - get
  - list
  - watch
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: weave-net
  namespace: kube-system
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: weave-net
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: weave-net
subjects:
- kind: ServiceAccount
  name: weave-net
  namespace: kube-system
```

The other changes are adding `serviceAccountName: weave-net` to the `PodSpec` of the `DaemonSet` and moving to the beta `Tolerations` field on the `PodSpec`.

@bboreham @lukemarsden @errordeveloper I can submit a new yaml file for this as a PR if you'd like

**Comment by luxas** [comment]
Another thing, we should also set a selinux option so the hostPath bindings are permitted:
`PodSpec.SecurityOptions.SELinuxOptions.Type=spc_t` should be set in the weave-kube.yaml file

x-ref: https://github.com/kubernetes/kubeadm/issues/107

**Comment by bboreham** [comment]
Note comment at https://github.com/weaveworks/flux/issues/400#issuecomment-282859843

> both alpha and beta APIs exist in 1.6, and a role created with v1alpha1 will also be accessible from the v1beta1 API

so I tried this, and it seemed to work at https://circleci.com/gh/weaveworks/weave/8146 (albeit the 'publish' phase failed on branch naming)

**Comment by bboreham** [comment]
Closed by #2801
